src/app/clusters/occupancy-sensor-server/OccupancySensingCluster.cpp

```diff
@@ -0,0 +1,342 @@
+        { mHasHoldTime, Attributes::HoldTime::kMetadataEntry },
+        { mHasHoldTime, Attributes::HoldTimeLimits::kMetadataEntry },
+        { mHasHoldTime && mFeatureMap.Has(Feature::kPassiveInfrared), Attributes::PIROccupiedToUnoccupiedDelay::kMetadataEntry },
+        { mHasHoldTime && mFeatureMap.Has(Feature::kUltrasonic), Attributes::UltrasonicOccupiedToUnoccupiedDelay::kMetadataEntry },
+        { mHasHoldTime && mFeatureMap.Has(Feature::kPhysicalContact),
+          Attributes::PhysicalContactOccupiedToUnoccupiedDelay::kMetadataEntry },
+    };
+
+    return listBuilder.Append(Span(OccupancySensing::Attributes::kMandatoryMetadata), Span(optionalAttributes));
+}
+
+DataModel::ActionReturnStatus OccupancySensingCluster::SetHoldTime(uint16_t holdTime)
+{
+    VerifyOrReturnError(mHasHoldTime, Protocols::InteractionModel::Status::UnsupportedAttribute);
+    VerifyOrReturnError(holdTime >= mHoldTimeLimits.holdTimeMin, Protocols::InteractionModel::Status::ConstraintError);
+    VerifyOrReturnError(holdTime <= mHoldTimeLimits.holdTimeMax, Protocols::InteractionModel::Status::ConstraintError);
+
+    if (mHoldTime == holdTime)
+    {
+        return DataModel::ActionReturnStatus::FixedStatus::kWriteSuccessNoOp;
+    }
+
+    mHoldTime = holdTime;
+
+    // If a timer is currently active, we need to adjust its duration to reflect the new hold time.
+    if (mTimerDelegate && mTimerDelegate->IsTimerActive(this))
+    {
+        // Calculate the time that has already elapsed since the timer was started.
+        System::Clock::Timestamp now           = mTimerDelegate->GetCurrentMonotonicTimestamp();
+        System::Clock::Timestamp elapsedTime   = now - mTimerStartedTimestamp;
+        System::Clock::Seconds16 newHoldTime16 = System::Clock::Seconds16(mHoldTime);
+
+        // If the elapsed time is already greater than or equal to the new hold time,
+        // the timer should have already fired. Cancel the current timer and immediately
+        // set the state to unoccupied.
+        if (elapsedTime >= newHoldTime16)
+        {
+            mTimerDelegate->CancelTimer(this);
+            DoSetOccupancy(false);
+        }
+        else
+        {
+            // Otherwise, restart the timer for the remaining duration.
+            System::Clock::Timeout remainingTime = newHoldTime16 - elapsedTime;
+            mTimerDelegate->StartTimer(this, remainingTime);
+        }
+    }
+
+    if (mDelegate)
+    {
+        mDelegate->OnHoldTimeChanged(holdTime);
+    }
+
+    if (mContext != nullptr)
+    {
+        CHIP_ERROR err =
+            mContext->attributeStorage.WriteValue({ mPath.mEndpointId, OccupancySensing::Id, Attributes::HoldTime::Id },
+                                                  { reinterpret_cast<const uint8_t *>(&mHoldTime), sizeof(mHoldTime) });
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(AppServer, "Failed to store holdTime in persistence: %" CHIP_ERROR_FORMAT, err.Format());
```

Comment:
Why log this? It's likely not recoverable. Suggest avoiding the log on embedded builds (perhaps this is another config?)